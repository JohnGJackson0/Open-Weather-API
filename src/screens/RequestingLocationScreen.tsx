import React, { useEffect, useState } from "react";
import { View, Text } from "react-native";
import * as Location from "expo-location";

export default function RequestingLocationScreen({ navigation }: any) {
  const [message, setMessage] = useState("");
  const [location, setLocation] = useState({});

  useEffect(() => {
    setMessage("waiting for geo location...");

    (async () => {
      let { status } = await Location.requestForegroundPermissionsAsync();
      console.log(status);
      if (status !== "granted") {
        setMessage("Permission to geo locate was denied");
        return;
      }
      setLocation((await Location.getCurrentPositionAsync({})).coords);
    })();
  }, []);

  useEffect(() => {
    console.log("The Location ", location);
    if (Object.keys(location).length !== 0) {
      navigation.reset({
        index: 0,
        routes: [{ name: "Home", params: { location: location } }],
      });
    }
  }, [location]);

  return (
    <View testID="RequestingLocationScreen">
      <Text>Follow the prompt to get permission to find your location.</Text>
      <Text>{message}</Text>
    </View>
  );
}
