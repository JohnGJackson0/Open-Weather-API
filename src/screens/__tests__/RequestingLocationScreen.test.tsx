import React from "react";
import { render, waitFor } from "@testing-library/react-native";
import RequestingLocationScreen from "../RequestingLocationScreen";

describe("Requesting Location Service", () => {
  it("renders instructions", () => {
    const fakeNavigator = {
      reset: jest.fn(),
    };

    const { getByText } = render(
      <RequestingLocationScreen navigation={fakeNavigator} />
    );

    getByText(/permission/);
  });
  it("It wont route with no permission accepted", () => {
    const fakeNavigator = {
      reset: jest.fn(),
    };
    render(<RequestingLocationScreen navigation={fakeNavigator} />);

    expect(fakeNavigator.reset).not.toHaveBeenCalled();
  });
  it("Should match the snapshot exactly", async () => {
    const fakeNavigator = {
      reset: jest.fn(),
    };

    const component = render(
      <RequestingLocationScreen navigation={fakeNavigator} />
    );

    expect(component).toMatchSnapshot();
  });
});
