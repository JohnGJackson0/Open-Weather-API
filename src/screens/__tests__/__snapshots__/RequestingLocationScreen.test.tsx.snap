// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Requesting Location Service Should match the snapshot exactly 1`] = `
<View
  testID="RequestingLocationScreen"
>
  <Text>
    Follow the prompt to get permission to find your location.
  </Text>
  <Text>
    waiting for geo location...
  </Text>
</View>
`;
